[WIP] TransformExecutor Tests

f3l1x98 opened the pull request:
Closes #415 

This PR adds tests for `TransformExecutor`.

f3l1x98 commented:
**Two things I have found:**
1. For the sake of creating a table as simple and fast as possible I added an constructor to the `Table` class. If I overlooked a reason as to why not do that please comment.
2. I wanted to add a test that would check if a row has been deleted because the expression could not be evaluated and I discovered that currently this only happens if there is no expression. The reason for that is that the `evaluateExpression` function throws assertion errors in case there is an issue which are not caught inside the `TransformExecutor#doExecuteTransform`. Is that intended, or should I add a try catch? (Because the logged error sounds like it should also handle the exceptions)

georg-schwarz commented on libs/execution/src/lib/transforms/transform-executor.spec.ts:
```diff
@@ -134,18 +134,24 @@ describe('Validation of TransformExecutor', () => {
+      expect(result.resultingColumn.values).toEqual(
+        expect.arrayContaining([21]),
+      );
+    } catch (e) {
+      expect(e).toEqual(undefined);
+    }
```
Why do we need that try-catch here?

f3l1x98 replied:
Strictly speaking we don't just added it that all tests have the same format.

georg-schwarz replied:
Then pls remove since it is kind of misleading

georg-schwarz commented on libs/execution/src/lib/transforms/transform-executor.spec.ts:
```diff
@@ -0,0 +1,371 @@
+
+    const transform = locator.getAstNode<TransformDefinition>(
+      document.parseResult.value,
+      'transforms@0',
```
Could you pls iterate over all transforms? Makes the test setup less brittle.

f3l1x98 replied:
That would most likely not change anything because the execution of the other transforms will most likely crash due to wrong input / execution context

f3l1x98 replied:
Discussed in person

georg-schwarz commented:
> **Two things I have found:**
> 
> 1. For the sake of creating a table as simple and fast as possible I added an constructor to the `Table` class. If I overlooked a reason as to why not do that please comment.
> 2. I wanted to add a test that would check if a row has been deleted because the expression could not be evaluated and I discovered that currently this only happens if there is no expression. The reason for that is that the `evaluateExpression` function throws assertion errors in case there is an issue which are not caught inside the `TransformExecutor#doExecuteTransform`. Is that intended, or should I add a try catch? (Because the logged error sounds like it should also handle the exceptions)

Could you post a link to the file?
Assertions should never fail, they point to a bug.
If a user led to the mistake, there should be an exception IMO.

f3l1x98 commented:
> > **Two things I have found:**
> > 
> > 1. For the sake of creating a table as simple and fast as possible I added an constructor to the `Table` class. If I overlooked a reason as to why not do that please comment.
> > 2. I wanted to add a test that would check if a row has been deleted because the expression could not be evaluated and I discovered that currently this only happens if there is no expression. The reason for that is that the `evaluateExpression` function throws assertion errors in case there is an issue which are not caught inside the `TransformExecutor#doExecuteTransform`. Is that intended, or should I add a try catch? (Because the logged error sounds like it should also handle the exceptions)
> 
> Could you post a link to the file? Assertions should never fail, they point to a bug. If a user led to the mistake, there should be an exception IMO.

I am talking about this commit (seems like I already added the try catch :sweat_smile: ): https://github.com/jvalue/jayvee/commit/28b5836b9c1363b4b02f04ca4332db8da80bc348

Because without this commit any exceptions thrown inside `evaluateExpression` will propagate through the whole stack up to the tests (or the system somewhere higher up in production).
Not sure why I wrote assertion errors instead of exceptions...
